joomla/joomla-cms
# [5.0]  tinyMCE b/c fixes

Pull Request for Issue #40622


### Summary of Changes

- @richard67 added a function to transform some J3/4 parameters to their new expected values for tinyMCE v6
- Replaced the same values in the installation SQL files
- Removed the template plugin as it's deprecated
- Introduce a fork of the template plugin, named `jtemplate`
- Patch both the builder for the above change
- Introduce a function `onAjaxTinymce()`, so that templates will be fetched on request rather than on the rendering of the editor (security + perf)
- Removed/renamed plugins and options to satisfy the changes of tinyMCE v6

### Testing Instructions

1. d/l and install a fresh copy of Joomla from the GH PR. Check that tinyMCE works as expected, particularly check the templates functionality.

2. Have a J4 installation and create some more `TinyMCE Sets`. Try to use all the buttons,menus,etc, and also have some html (or txt) files as templates in the `cassiopeia/html/tinymce`. UPGRADE to J5 using the `Custom URL` in the `administrator/index.php?option=com_config&view=component&component=com_joomlaupdate&path=` with a vaule `https://ci.joomla.org/artifacts/joomla/joomla-cms/5.0-dev/40626/downloads/65966/pr_list.xml`. Enure that the configuarations are identical and that the templates are still working.


### Actual result BEFORE applying this Pull Request

Broken Upgrade

### Expected result AFTER applying this Pull Request

No breaks

### Link to documentations
Please select:


@brianteeman @HLeithner @laoneo
